libclc/ptx-nvidiacl/libspirv/images/image.cl

```diff
@@ -973,3 +975,1511 @@ _CLC_DEF int3 _Z22__spirv_ImageQuerySizeIDv3_i14ocl_image3d_roET_T0_(
+  short4 ret;
+  ret = __nvvm_suld_2d_v4i16_clamp_s(imageHandle, x, y);
+  short2 b;
+  b.x = ret.x;
+  b.y = ret.y;
+  return b;
+}
+short2 __nvvm_suld_3d_v2i16_clamp_s(long imageHandle, int x, int y, int z) {
+  short4 ret;
+  ret = __nvvm_suld_3d_v4i16_clamp_s(imageHandle, x, y, z);
+  short2 b;
+  b.x = ret.x;
+  b.y = ret.y;
+  return b;
+}
+
+// unsigned short
+unsigned short __nvvm_suld_1d_t16_clamp_s(long imageHandle, int coord) {
+  return as_ushort(__nvvm_suld_1d_i16_clamp_s(imageHandle, coord));
+}
+unsigned short __nvvm_suld_2d_t16_clamp_s(long imageHandle, int x, int y) {
+  return as_ushort(__nvvm_suld_2d_i16_clamp_s(imageHandle, x, y));
+}
+unsigned short __nvvm_suld_3d_t16_clamp_s(long imageHandle, int x, int y, int z) {
+  return as_ushort(__nvvm_suld_3d_i16_clamp_s(imageHandle, x, y, z));
+}
+ushort2 __nvvm_suld_1d_v2t16_clamp_s(long imageHandle, int coord) {
+  return as_ushort2(__nvvm_suld_1d_v2i16_clamp_s(imageHandle, coord));
+}
+ushort2 __nvvm_suld_2d_v2t16_clamp_s(long imageHandle, int x, int y) {
+  return as_ushort2(__nvvm_suld_2d_v2i16_clamp_s(imageHandle, x, y));
+}
+ushort2 __nvvm_suld_3d_v2t16_clamp_s(long imageHandle, int x, int y, int z) {
+  return as_ushort2(__nvvm_suld_3d_v2i16_clamp_s(imageHandle, x, y, z));
+}
+ushort4 __nvvm_suld_1d_v4t16_clamp_s(long imageHandle, int coord) {
+  return as_ushort4(__nvvm_suld_1d_v4i16_clamp_s(imageHandle, coord));
+}
+ushort4 __nvvm_suld_2d_v4t16_clamp_s(long imageHandle, int x, int y) {
+  return as_ushort4(__nvvm_suld_2d_v4i16_clamp_s(imageHandle, x, y));
+}
+ushort4 __nvvm_suld_3d_v4t16_clamp_s(long imageHandle, int x, int y, int z) {
+  return as_ushort4(__nvvm_suld_3d_v4i16_clamp_s(imageHandle, x, y, z));
+}
+
+// signed char
+short __nvvm_suld_1d_i8_clamp_s_helper(long,
+                                       int) __asm("llvm.nvvm.suld.1d.i8.clamp");
+signed char __nvvm_suld_1d_i8_clamp_s(long imageHandle, int coord) {
+  return as_char(
+      (signed char)__nvvm_suld_1d_i8_clamp_s_helper(imageHandle, coord));
+}
+short __nvvm_suld_2d_i8_clamp_s_helper(long, int,
+                                       int) __asm("llvm.nvvm.suld.2d.i8.clamp");
+signed char __nvvm_suld_2d_i8_clamp_s(long imageHandle, int x, int y) {
+  return as_char(
+      (signed char)__nvvm_suld_2d_i8_clamp_s_helper(imageHandle, x, y));
+}
+short __nvvm_suld_3d_i8_clamp_s_helper(long, int, int,
+                                       int) __asm("llvm.nvvm.suld.3d.i8.clamp");
+signed char __nvvm_suld_3d_i8_clamp_s(long imageHandle, int x, int y, int z) {
+  return as_char(
+      (signed char)__nvvm_suld_3d_i8_clamp_s_helper(imageHandle, x, y, z));
+}
+short2 __nvvm_suld_1d_v2i8_clamp_s_helper(long,
+                                       int) __asm("__clc_llvm_nvvm_suld_1d_v2i8_clamp");
+char2 __nvvm_suld_1d_v2i8_clamp_s(long imageHandle, int coord) {
+  short2 a = __nvvm_suld_1d_v2i8_clamp_s_helper(imageHandle, coord);
+  char2 ret;
+  ret.x = (char) a.x;
```

Comment:
I've formatted all of our non-macro code and added some whitespace where the code was a bit crowded. I think it made some sections of the code slightly less readable though, but if we want to enforce clang-formatting with line length limit then that's how it will turn out.

Reply:
I would prefer to have clang-format on by default on all the code, with no modifications; there is already a rules file that [can be used](https://github.com/intel/llvm/blob/sycl/.clang-format). In my experience, it is always beneficial in a long run, as in one less problem to think about.